cmd/thanos/compact.go

```diff
@@ -285,6 +306,7 @@ func runCompact(
 		return errors.Wrap(err, "clean working downsample directory")
 	}
 
+	blocksCleaner := compact.NewBlocksCleaner(logger, bkt, deleteDelay, blocksCleaned)
```

Comment:
Hm, I was expecting this to take `ignoreDeletionMarkFilter`

Reply:
Sure, that is a really good feedback point. I missed that. Thanks for pointing this out.

Reply:
made this change.

Reply:
Please note that we require two `ignoreDeletionMarkFilter` filters in compact now:
a. First one will have delay as deleteDelay/2, which will be used in compact
b. One which has delay as deleteDelay, which will be used in blocks_cleaner

This is because we filter blocks based on deleteDelay/2 duration, the blocks_cleaner has to clean after deleteDelay.

Reply:
As per my latest changes, I am using blocks cleaner only and not using two filters for this.

Reply:
Marking this as resolved since this is taken care of and latest feedback doesn't have new suggestions for this.